napari/layers/base/base.py

```diff
@@ -1295,6 +1291,38 @@ def _world_to_displayed_data_ray(
         vector_data_ndisplay /= np.linalg.norm(vector_data_ndisplay)
         return vector_data_ndisplay
 
+    def _world_to_data_dims_displayed(
+        self, dims_displayed: List[int], ndim_world: int
+    ) -> np.ndarray:
+        """Convert the displayed dims in world coordinates to the data coordinates.
+
+        This accounts for differences in dimensionality of the world and the data.
+        for example a global order of [2, 1, 0, 3] -> [0, 1] for a layer that
+        only has two dimensions or -> [1, 0, 2] for a layer with three as that
+        corresponds to the relative order of the last two and three dimensions
+        respectively
```

Comment:
I think I understand the need for this but I found this docstring quite hard to parse - I don't think I understand it well enough to have a better suggestion so don't consider it blocking...

Am I right in thinking that this is needed only when there is both a mix in dimensionality and the dims have been rolled?

Reply:
Yes, this is for determining which dimensions of the layer data are being displayed. It is most important for the case when the layer has fewer dimensions than the world, but it works when layer ndim is the same as world ndim. I can try to rework the docstring.

Reply:
I've updated the docstring a bit. Let me know if there are specific things that are still unclear.

Reply:
looks great! Great job figuring this out, my interpretation of how this *should* work was actually wrong... it's worth noting that this behaviour may change if we move forward with changes to how dims are handled discussed in #3848